src/Telegram/Endpoints/AvailableMethods.php

```diff
@@ -73,13 +73,32 @@ public function close(): ?bool
      * @see https://core.telegram.org/bots/api#sendmessage
      * @param  string  $text Text of the message to be sent, 1-4096 characters after entities parsing
      * @param  array|null  $opt
-     * @return Message|null
+     * @return Message|Message[]|null
      */
-    public function sendMessage(string $text, ?array $opt = []): ?Message
+    public function sendMessage(string $text, ?array $opt = []): Message|array|null
     {
         $chat_id = $this->chatId();
         $required = compact('text', 'chat_id');
-        return $this->requestJson(__FUNCTION__, array_merge($required, $opt), Message::class);
+        $parameters = array_merge($required, $opt);
+
+        if ($this->config['chunk_message'] ?? false) {
+            $chunks = $this->chunkText($text);
+            $reply_markup = $parameters['reply_markup'] ?? null;
+            unset($parameters['reply_markup']);
+
+            $messages = [];
+            foreach ($chunks as $index => $chunk) {
+                if ($index === count($chunks) - 1) {
+                    $parameters['reply_markup'] = $reply_markup;
+                }
+                $parameters['text'] = $chunk;
+
```

Comment:
Need to pass sleep for 1 sec. parameter to avoid "429 Rate Limiting" error of 1 message per second 
https://core.telegram.org/bots/faq#my-bot-is-hitting-limits-how-do-i-avoid-this

Reply:
ok but why exactly 1 second?

Reply:
Here is not needed the `sleep` call, Telegram allows short burst as they written in their FAQs you linked, and this case falls in that exact use case. If your message is going to be split in more that 3-4 messages (and so requests), than sending a file should be more appropriate.